temporian/implementation/numpy/operators/test/cast_test.py

```diff
@@ -0,0 +1,245 @@
+                    [TRYOLABS_SHOP, MATE_ID, 1.0, 15.0, below_int32, 2],
+                    [TRYOLABS_SHOP, MATE_ID, 2.0, 16, 3, 0],
+                    [TRYOLABS_SHOP, BOOK_ID, 1.0, 10, 4, 100],
+                    [GOOGLE_SHOP, BOOK_ID, 1.0, 0, 5, 3.2],
+                    [GOOGLE_SHOP, BOOK_ID, 2.0, above_int64, 6, 1],
+                    [GOOGLE_SHOP, PIXEL_ID, 0.0, -1, -8, -3.5],
+                    [GOOGLE_SHOP, PIXEL_ID, 1.0, below_int64, 8, -2],
+                ],
+                columns=[
+                    "store_id",
+                    "product_id",
+                    "timestamp",
+                    "float_64",
+                    "int_64",
+                    "str",
+                ],
+                # Even more tricky: these columns won't match their type
+            ).astype({"float_64": np.float32, "int_64": np.int32}),
+            index_names=["store_id", "product_id"],
+        )
+
+        self.sampling = Sampling(
+            [("store_id", dtype.INT32), ("product_id", dtype.INT64)]
+        )
+        self.input_event = Event(
+            [
+                Feature("float_64", dtype.FLOAT64),
+                Feature("int_64", dtype.INT64),
+                Feature("str", dtype.STRING),
+            ],
+            sampling=self.sampling,
+            creator=None,
+        )
+
+    def test_cast_by_feature(self) -> None:
+        """Test correct casting by feat. names, without overflow check."""
+
+        operator = CastOperator(
+            event=self.input_event,
+            to={
+                "float_64": dtype.FLOAT32,
+                "int_64": dtype.INT32,
+                "str": dtype.FLOAT64,
+            },
+            check_overflow=False,
+        )
+
+        cast_implementation = CastNumpyImplementation(operator)
+
+        operator_output = cast_implementation.call(event=self.numpy_in_event)
+
+        self.assertTrue(self.numpy_expected_event == operator_output["event"])
+
+    def test_cast_by_dtype(self) -> None:
+        """Test correct casting by origin dtype, without overflow check."""
+
+        operator = CastOperator(
+            event=self.input_event,
+            to={
+                dtype.FLOAT64: dtype.FLOAT32,
+                dtype.INT64: dtype.INT32,
+                dtype.STRING: dtype.FLOAT64,
+                dtype.INT32: dtype.INT64,  # This one should have no effect
+            },
+            check_overflow=False,
+        )
+
+        cast_implementation = CastNumpyImplementation(operator)
+
+        operator_output = cast_implementation.call(event=self.numpy_in_event)
+
+        self.assertTrue(self.numpy_expected_event == operator_output["event"])
+
+    def test_cast_mixed_keys(self) -> None:
+        """Test correct casting, some cols by origin dtype,
+        others by feature name, without overflow check.
+        """
+
+        operator = CastOperator(
+            event=self.input_event,
+            to={
```

Comment:
I just feel this may be confusing in the future or error-prone.

What do you think of the possibility of having two attributes on CastOperator:

1. `from_name` or `from_feature_name`
2. `from_dtype`

With this, we have two explicit dictionaries. It may make the code cleaner and it's more explicit for the user.

I'm open to suggestions or critics.

Reply:
Not sure about this one.
IMO the usage is more concise this way, and it's also pretty explicit. It would need some sort of precedence anyway with two separate dicts (if a feature matches feature_name and dtype), or raise error with two arguments at the same time. Also, `from_dtype` could be a dtype or a dict.

But I'm open to what the majority feels as more intuitive. Opinions @achoum @getchebarne ?